packages/account/src/Sections/Profile/PersonalDetails/personal-details.jsx

```diff
@@ -298,9 +298,41 @@ export class PersonalDetailsForm extends React.Component {
             }
         }
 
+        if (Object.keys(errors).length > 0) {
+            this.setState({ errors: true });
+        } else {
+            this.setState({ errors: false });
+        }
+
         return errors;
     };
 
+    getWarningMessages = values => {
+        const warnings = {};
+        const active_errors = this.state.errors;
+        const account_object = this.props;
+        const residence_list_array = Object.values(account_object.residence_list);
+
+        const filter_tin_regex = residence_list_array.filter(residence => {
+            return residence.text === values.tax_residence && residence.tin_format;
+        });
+
+        const tin_regex = filter_tin_regex[0]?.tin_format?.[0];
+        const test_tin = new RegExp(tin_regex).test(values.tax_identification_number);
+        const valid_country_tin = tin_regex ? test_tin : true;
+
+        if (!active_errors) {
+            if (values.tax_identification_number) {
+                if (!valid_country_tin) {
+                    const warning_message =
+                        'This Tax Identification Number (TIN) is invalid. You may continue using it, but to facilitate future payment processes, valid tax information will be required.';
+                    warnings.tax_identification_number = localize(warning_message);
```

Comment:
we can't use the `localize` function like this. we should inject the string directly into `localize` and not as a variable because it is done on build time, (not run time)

Reply:
Aaah ofcourse I see now. I will try to keep this in mind. Thanks! Fixed this now aswell.